### DISCUSSION TITLE
SECTION-073 リスト18.14のinterface{}ってanyとどう使い分けているのでしょうか?

### DISCUSSION THREAD

**Discussion by nnabeyang** [budougumi0617/go_todo_app#72] [Q&A] [answered]
次の部分を見ると、destのような書き換えるものはanyは使わない習慣があるのかなとも思ったのですが、
https://github.com/budougumi0617/go_todo_app/blob/main/_chapter18/section74/store/repository.go#L71-L72
次のようなargでもinterface{}の場合があり、よく分かりませんでした。
https://github.com/budougumi0617/go_todo_app/blob/main/_chapter18/section74/store/repository.go#L64

anyはinterface{}のエイリアスだと認識しているので、何かしら使い分けの習慣があるのかと思ったのですが、どうでしょうか?

**Comment by budougumi0617** [comment, marked as answer]
ご質問ありがとうございます。鋭い指摘ですね…!

`repository.go` の `any` と `interface{}` の使い分けは標準パッケージ由来のシグネチャか、 `github.com/jmoiron/sqlx` パッケージ由来のシグネチャかによって変わっています。
他のシーンでも適用できる一般化できるルールに則っているわけではなかったりします。

`ExecContext` などは標準パッケージの定義なのでany対応されてシグネチャが宣言されています。
https://pkg.go.dev/database/sql#DB.ExecContext
対して、 `SelectContext` などのsqlx特有のものは大元が `interface{}` のままなのでsqlxに合わせて `interface{}` にしてあるといった感じです。
https://pkg.go.dev/github.com/jmoiron/sqlx#DB.SelectContext

（といいつつ今見るとsqlxにしかないQueryxContextでも `any` になっていたりしますね 🙇

**Reply by nnabeyang** [reply]
@budougumi0617 なるほど。空気感が少し分かりました。ありがとうございます。
